GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java

```diff
@@ -665,27 +671,73 @@ private void writeAnswer(
                 break;
 
             case NUMBER:
-                val = value.trim();
-                cells.add(val);
+                cells.add(value);
                 break;
 
+            // Two different formats:
+            // old: Foo|Bar|Baz (never contains code)
+            // new: [{"name": "Foo", "code": "A"}, {"name": "Bar", "code": "B"} ... ]
             case CASCADE:
+                List<Map<String, String>> parts = new ArrayList<>();
+
+                if (value.startsWith("[")) {
+                    try {
+                        parts = OBJECT_MAPPER.readValue(value,
+                                new TypeReference<List<Map<String, String>>>() {
+                                });
+                    } catch (IOException e) {
+                        log.warn("Unable to parse CASCADE response - " + value, e);
+                    }
+                } else {
+                    for (String part : value.split("\\|")) {
+                        Map<String, String> m = new HashMap<>();
+                        m.put("name", part);
+                        parts.add(m);
+                    }
+                }
+
                 if (useQuestionId) {
-                    String cellVal = value.trim();
+                    // +------------+------------+-----
+                    // |code1:value1|code2:value2| ...
+                    // +------------+------------+-----
+
                     int levelCount = questionDto.getLevelNames().size();
-                    List<String> parts = new ArrayList<String>(Arrays.asList(cellVal
-                            .split("\\|", levelCount)));
                     int padCount = levelCount - parts.size();
 
-                    for (int p = 0; p < padCount; p++) { // padding
-                        parts.add("");
+                    for (Map<String, String> map : parts) {
+                        String code = map.get("code");
+                        String name = map.get("name");
+                        String nodeVal = (code == null ? "" : code + ":") + name;
+
+                        if (cells.size() == levelCount) {
```

Comment:
This is the issue where the number of cascade levels defined is the question < levels in the response. We need to do something here  otherwise we "spill over" into the columns of other questions. Our options are limited:

* don't display more levels than the question specifies => we lose some of the response
* fall back to pipe separated rendering (this is the current approach)